src/styles/StyleUtils.js

```diff
@@ -300,6 +300,24 @@ function getWidthStyle(width) {
     };
 }
 
+/**
+ * Returns auto grow height text input style
+ *
+ * @param {Number} textInputHeight
+ * @param {Number} maxHeight
+ * @returns {Object}
+ */
+function getAutoGrowHeightInputStyle(textInputHeight, maxHeight) {
+    if (textInputHeight > maxHeight) {
+        return styles.overflowAuto;
+    }
+
+    return {
+        ...styles.overflowHidden,
+        height: maxHeight,
```

Comment:
`padding` and `borderwidth` were not considered in maxHeight. It caused https://github.com/Expensify/App/issues/18918 issue regression which is solved by https://github.com/Expensify/App/pull/19582/files